src/scripts/main.js

```diff
@@ -1,3 +1,45 @@
 'use strict';
 
 // write code here
+const headline = document.querySelector('thead');
+
+function getCleanedValue(row, index) {
+  const cells = row.querySelectorAll('td')[index];
+
+  if (!cells) {
+    return '';
+  }
+
+  const cleanedCell = cells.textContent.trim().replace(/[$,]/g, '');
+
+  return isNaN(cleanedCell) ? cleanedCell : parseInt(cleanedCell);
+}
```

Comment:
Code logic issue: The function `getCleanedValue` is trying to access the `textContent` of a `td` element, but there's a chance that the `querySelectorAll('td')[index]` might return undefined. It's better to check if the `cells` is defined before accessing its `textContent` property.